packages/moonstone/MoonstoneDecorator/MoonstoneDecorator.js

```diff
@@ -118,17 +120,113 @@ const MoonstoneDecorator = hoc(defaultConfig, (config, Wrapped) => {
 	const Decorator = class extends React.Component {
 		static displayName = 'MoonstoneDecorator';
 
+		static contextTypes = contextTypes
+
+		static childContextTypes = contextTypes
+
+		static propTypes =  /** @lends moonstone/MoonstoneDecorator.AccessibilityDecorator.prototype */ {
+			/**
+			 * Enables additional features to help users visually differentiate components.
+			 * The UI library will be responsible for using this information to adjust
+			 * the components' contrast to this preset.
+			 *
+			 * @type {Boolean}
+			 * @public
+			 */
+			highContrast: PropTypes.bool,
+
+			/**
+			 * Set the goal size of the text. The UI library will be responsible for using this
+			 * information to adjust the components' text sizes to this preset.
+			 * Current presets are `'normal'` (default), and `'large'`.
+			 *
+			 * @type {String}
+			 * @default 'normal'
+			 * @public
+			 */
+			textSize: PropTypes.oneOf(['normal', 'large'])
+		}
+
+		static defaultProps = {
+			highContrast: false,
+			textSize: 'normal'
+		}
+
+		getChildContext () {
+			return {
+				Subscriber: this.publisher.getSubscriber()
+			};
+		}
+
+		componentWillMount () {
```

Comment:
Just a quick note, since we know `componentWillMount` is eventually going away, should we move this to `componentDidMount` or otherwise update it?

Reply:
Not sure. If we're looking ahead we may not even need this change since we could use the new `Context API`. I guess I could test it to see if perf is different now as well.